src/D2L.CodeStyle.Analyzers/Immutability/MutabilityInspector.cs

```diff
@@ -291,6 +291,17 @@ HashSet<ITypeSymbol> typeStack
 
 			SemanticModel model = m_compilation.GetSemanticModel( expr.SyntaxTree );
 
+			if( expr.Kind() == SyntaxKind.SimpleLambdaExpression ||

```

Comment:
nit: you could move this above the `GetSemanticModel` (for performance) and then merge with the other `if` and create a `switch`

Reply:
Yep. It was left down there from when I was playing around with the `DataFlowAnalysis` stuff. I'll fix that

Reply:
Oh yea, cool